src/node/internal/internal_fs_promises.ts

```diff
@@ -0,0 +1,324 @@
+// Copyright (c) 2017-2022 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+//
+// Copyright Joyent, Inc. and other Node contributors.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to permit
+// persons to whom the Software is furnished to do so, subject to the
+// following conditions:
+//
+// The above copyright notice and this permission notice shall be included
+// in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN
+// NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+// OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE
+// USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+/* eslint-disable @typescript-eslint/require-await,@typescript-eslint/no-unused-vars */
+
+import {
+  getOptions,
+  copyObject,
+  getValidatedPath,
+  kMaxUserId,
+  validateCpOptions,
+  toUnixTimestamp,
+  validateRmdirOptions,
+} from 'node-internal:internal_fs_utils';
+import * as constants from 'node-internal:internal_fs_constants';
+import {
+  parseFileMode,
+  validateInteger,
+  validateBoolean,
+  validateAbortSignal,
+  validateOneOf,
+} from 'node-internal:validators';
+import type { RmDirOptions, CopyOptions } from 'node:fs';
+
+type FilePath = string | URL;
+
+export async function access(
+  _path: FilePath,
+  _mode: number = constants.F_OK
+): Promise<void> {
+  throw new Error('Not implemented');
+}
+
+export async function appendFile(
+  _path: FilePath,
+  _data: unknown,
+  options: Record<string, unknown>
+): Promise<void> {
+  options = getOptions(options, { encoding: 'utf8', mode: 0o666, flag: 'a' });
+  options = copyObject(options);
+  options.flag ||= 'a';
```

Comment:
Do we need to do some processing if we throw at the end anyway?

Reply:
Well this is going to change, since I'll be implementing node:fs. This is just a stub and some input validation to prepare for the actual work.

Reply:
Should this be a DRAFT PR at the moment?

Reply:
There are still todos on the description. But I don't plan to add the actual implementation on this PR.